pkg/git/provider.go

```diff
@@ -23,7 +25,18 @@ func NewGitProvider(ctx context.Context, providerType ProviderType, remoteURL, t
 	switch providerType {
 	case ProviderTypeAzdo:
 		return NewAzdoGITProvider(ctx, remoteURL, token)
+	case ProviderTypeUnknown:
+		return nil, fmt.Errorf("unknown provider type")
 	default:
 		return nil, fmt.Errorf("unknown provider type: %s", providerType)
 	}
 }
+
+func StringToProviderType(p string) ProviderType {
+	switch strings.ToLower(p) {
+	case "azdo":
+		return ProviderTypeAzdo
+	default:
```

Comment:
I would consider removing the unknown type and returning an error here unless there is a specific use for it.

Reply:
Removed!